### PR TITLE
feat(update-authors): add .mailmap support for coauthors

### PR DESCRIPTION
**Pull request by belochub**
Refs: https://github.com/metarhia/metatests/pull/208

### CODE REVIEW

File: bin/update-authors.js [hunk 1]
```diff
@@ -121,7 +120,24 @@ linereader.on('line', line => {
...
 
+const mailmapAuthors = async () => {
+  const mailmappedAuthors = new Set();
+  const exec = util.promisify(childProcess.exec);
+  for (const author of authors) {
+    const { stdout: contact } = await exec(
+      `git check-mailmap ${JSON.stringify(author)}`
+    );
+    mailmappedAuthors.add(contact.trim());
+  }
```
Comment (lundibundi) [inline bin/update-authors.js:131]: Nit, perhaps:
```js
authors = await Promise.all(
  authors.map(author => exec(`git check-mailmap ${JSON.stringify(author)}`))
).then(res => new Set(res.map(contact => contact.trim())));
```

**Review by lundibundi — APPROVED** [review]

File: bin/update-authors.js (hunk 1 above)
Comment (lundibundi) [reply bin/update-authors.js:131]: OR
```js
const contacts = await Promise.all(
  authors.map(author => exec(`git check-mailmap ${JSON.stringify(author)}`))
);
authors = new Set(res.map(contact => contact.trim()));

```

Comment (belochub) [reply bin/update-authors.js:131]: That wouldn't work, since `authors` is a `Set` object.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/tools/commit/cecae4b273b0c9859731eb89da29ce6d965b67bb.

**Pull request closed without merging**